generator/src/main/java/com/scottlogic/deg/generator/validators/GenerationConfigValidator.java

```diff
@@ -21,4 +40,48 @@ public ValidationResult validateCommandLine(GenerationConfig config) {
 
         return validationResult;
     }
-}
\ No newline at end of file
+
+    public ValidationResult validateCommandLinePostProfile(Profile profile) {
+        ArrayList<String> errorMessages = new ArrayList<>();
+        ValidationResult validationResult = new ValidationResult(errorMessages);
+
+        if (configSource.shouldViolate()) {
+            checkViolationGenerateOutputTarget(errorMessages, outputTarget, profile.rules.size());
+        } else {
+            checkGenerateOutputTarget(errorMessages, outputTarget);
+        }
+
+        return validationResult;
+    }
+
+    private void checkGenerateOutputTarget(ArrayList<String> errorMessages,
+                                           OutputTarget outputTarget) {
+        if (outputTarget instanceof FileOutputTarget) {
+            if (fileUtils.isDirectory(outputTarget)) {
+                errorMessages.add(
+                    "Invalid Output - target is a directory, please use a different output filename");
+            } else if (!configSource.overwriteOutputFiles() && fileUtils.exists(outputTarget)) {
+                errorMessages.add(
+                    "Invalid Output - file already exists, please use a different output filename or use the --overwrite option");
+            }
+        }
+    }
+
+    private void checkViolationGenerateOutputTarget(ArrayList<String> errorMessages,
+                                                    OutputTarget outputTarget, int ruleCount) {
+        if (outputTarget instanceof FileOutputTarget) {
+            if (!fileUtils.exists(outputTarget)) {
+                errorMessages.add(
+                    "Invalid Output - output directory must exist. please enter a valid directory name");
```

Comment:
Use correct grammar in error messages, capital P for please. The same is repeated in the other messages

Reply:
Capital's aren't usually used after a comma.

Reply:
fixed